### PR TITLE
Added Kiali chart

### PR DESCRIPTION
**Pull request by a-thaler**
**Description**

Changes proposed in this pull request:

- moved kiali into own chart
- enabled kcproxy for authentication

Fixes https://github.com/kyma-project/kyma/issues/4760 and https://github.com/kyma-project/kyma/issues/4918

### CODE REVIEW

File: resources/kiali/values.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    passphrase: ""
+  
+global:
+  isLocalEnv: false
+  containerRegistry:
+    path: eu.gcr.io/kyma-project
+  alpine_net:
+    name: alpine-net
+    dir: develop
+    version: 149967d0
```
Comment (Demonsthere) [inline resources/kiali/values.yaml]: ad quotes to the version string or in some cases when the version has only number helm will have problem with interpreting it

File: resources/kiali/templates/virtualservice.yaml [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
   labels:
     app: {{ template "kiali.name" . }}
-    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    chart: {{ template "kiali.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
 spec:
   gateways:
-  - {{ .Values.global.istio.gateway.name }}
+  - kyma-gateway
```
Comment (Demonsthere) [inline resources/kiali/templates/virtualservice.yaml]: Better to use a variable here as we have the name of the gateway set as variable also

File: resources/kiali/templates/tests/test-kiali-connection.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "false"
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: "{{ template "kiali.fullname" . }}-test"
+        image: {{ .Values.global.containerRegistry.path }}/{{ .Values.global.alpine_net.dir }}/{{ .Values.global.alpine_net.name }}:{{ .Values.global.alpine_net.version }}
+        imagePullPolicy: "{{ .Values.global.imagePullPolicy }}"
+        command: ['curl']
```
Comment (Demonsthere) [inline resources/kiali/templates/tests/test-kiali-connection.yaml]: If the test only curl kiali to see if it alive, better remove the test altogether and change this to a readiness check

File: resources/kiali/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  template:
+    metadata:
+      name: {{ template "kiali.name" . }}
+      labels:
+        app: {{ template "kiali.name" . }}
+        chart: {{ template "kiali.chart" . }}
+        heritage: {{ .Release.Service }}
+        release: {{ .Release.Name }}
+      annotations:
+        sidecar.istio.io/inject: "false"
```
Comment (Demonsthere) [inline resources/kiali/templates/deployment.yaml]: Any reason we need to disable istio sidecar?

Comment (a-thaler) [reply resources/kiali/templates/deployment.yaml]: I think all that observability tools should not be inside the service-mesh, no application should have a need to talk to it.
it calls prometheus only which is also outside the mesh. Have no better argument. Do you have an argument why it should be inside?

File: resources/kiali/templates/tests/test-kiali-connection.yaml (hunk 3 above)
Comment (a-thaler) [reply resources/kiali/templates/tests/test-kiali-connection.yaml]: I mainly want to assure that the full exposure is working as here most of the problems occur. 
I mainly adopted the test from the kiali chart itself and I think we should extend it in future to check that basic functionality is availble.

File: resources/kiali/values.yaml (hunk 1 above)
Comment (a-thaler) [reply resources/kiali/values.yaml]: done

File: resources/kiali/templates/virtualservice.yaml (hunk 2 above)
Comment (a-thaler) [reply resources/kiali/templates/virtualservice.yaml]: done

**Comment by a-thaler** [comment]
I re-added the kiali subchart to istio for smooth upgrades of istio. However, I still don't like the idea, a gitignore sounds much better. no-code is the best code.
@Demonsthere please check again

File: resources/kiali/README.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+[Kiali](http://kiali.io) is a spyglass for the istio service mesh and provides answers to the questions:
+What microservices are part of my Istio service mesh? How are they connected? How are they performing?
+
+## Introduction
+
+This chart installs Kiali on a Kyma cluster.
+
+## Prerequisites
+
+- The `monitoring` chart must be installed in order to have Kiali functional
```
Comment (bszwarc) [inline resources/kiali/README.md]:
```suggestion

-  Install the Monitoring chart to make Kiali work.
```

File: resources/kiali/README.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+- The `jaeger` chart should be installed to have the Jaeger integration of Kiali functional
```
Comment (bszwarc) [inline resources/kiali/README.md]:
```suggestion
- Install the Jaeger chart to integrate Jaeger with Kiali. 
```
Can we simply say `integrate` here or `to make Jaeger integration with Kiali work` sounds better? I would suggest the shorter version, but I don't want to change the meaning :)

File: resources/kiali/README.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+- The `logging` chart should be installed to have the Grafana integration of Kiali functional
```
Comment (bszwarc) [inline resources/kiali/README.md]:
```suggestion
- Install the Logging chart to integrate Grafana with Kiali.
```
Also the same question as above.

File: resources/kiali/README.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+
+## Installing the Chart
+
+To install the chart with the release name `kiali` in namespace kyma-system call:
```
Comment (bszwarc) [inline resources/kiali/README.md]:
```suggestion
To install the chart with the release name `kiali` in the `kyma-system` Namespace, run:
```

File: resources/kiali/README.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+    ```
+    $ helm install kiali --name kiali --namespace kyma-system
+    ```
+
+## Uninstalling the Chart
```
Comment (bszwarc) [inline resources/kiali/README.md]:
```suggestion
### Unistallation
```

File: resources/kiali/README.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+To uninstall/delete the `kiali` release but continue to track the release:
```
Comment (bszwarc) [inline resources/kiali/README.md]:
```suggestion
To uninstall/delete the `kiali` release but continue tracking the release, run:
```
If we can still track the release maybe we can simply have `unistall` here without the `delete`?

File: resources/kiali/README.md [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    ```
+    $ helm delete kiali
+    ```
+
+To uninstall/delete the `kiali` release completely and make its name free for later use:
```
Comment (bszwarc) [inline resources/kiali/README.md]:
```suggestion
To uninstall/delete the `kiali` release completely and make its name free for further use, run:
```
Maybe we can have `delete completely` without `unistall` here to distinguish it from the previous one?

**Pull request merged**